src/components/Counter.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { State } from '../types/State';
+import { Props } from '../types/Props';
+
+export class Counter extends React.Component<Props, State> {
+  state: State = {
+    number: 0,
+  };
+
+  addOne = () => {
+    this.setState((prevNumber) => ({
```

Comment:
```suggestion
    this.setState((prevState) => ({
```
Это весь предыдущий объект state, не только Number